WebDriverAgentLib/Categories/XCUIApplication+FBHelpers.m

```diff
@@ -108,4 +108,19 @@ + (NSDictionary *)accessibilityInfoForElement:(XCElementSnapshot *)snapshot
   return info;
 }
 
+-(void) fb_resolve
+{
+  if (self.fb_isActivateSupported) {
+    
+    if(self.state != XCUIApplicationStateRunningForeground)
+    {
+      [self activate];
```

Comment:
there are fb_activate/fb_state endpoints

Reply:
Yes but they call fb_isActivateSupported again. It seems like a wasted check since the if condition has already established that.

Reply:
I could use fb_activate and fb_state with try catches blocks, removing the need to call fb_isActivateSupported

Reply:
@mykola-mokhnach changed to us fb_activate/fb_state within a try/catch block removing the need for the if statement checking fb_isActivateSupported